src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/Target_X64/DelayLoadHelperThunk.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Internal.Text;
+using Internal.TypeSystem;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    /// <summary>
+    /// This node emits a thunk calling DelayLoad_Helper with a given instance signature
+    /// to populate its indirection cell.
+    /// </summary>
+    public partial class DelayLoadHelperThunk
+    {
+        protected override void EmitCode(NodeFactory factory, ref X64.X64Emitter instructionEncoder, bool relocsOnly)
+        {
+            // lea rax, [pCell]
+            instructionEncoder.EmitLEAQ(X64.Register.RAX, _instanceCell);
+            if (!relocsOnly)
+            {
+                // push table index
+                instructionEncoder.EmitPUSH((sbyte)_instanceCell.Table.IndexFromBeginningOfArray);
```

Comment:
Should this be a checked cast so that we don't have to debug potentially annoying runtime failures?